odoo/tools/sql.py

```diff
@@ -123,12 +123,20 @@ def constraint_definition(cr, constraintname):
     cr.execute("SELECT pg_get_constraintdef(oid) FROM pg_constraint WHERE conname=%s", (constraintname,))
     return cr.fetchone()[0] if cr.rowcount else None
 
+def constraint_comment(cr, constraintname):
+    """ Return the given constraint's comment. """
+    cr.execute("SELECT description FROM pg_description d JOIN pg_constraint c "
+               "ON c.oid = d.objoid AND conname = %s;", (constraintname,))
+    return cr.fetchone()[0] if cr.rowcount else None
+
 def add_constraint(cr, tablename, constraintname, definition):
     """ Add a constraint on the given table. """
     query = 'ALTER TABLE "{}" ADD CONSTRAINT "{}" {}'.format(tablename, constraintname, definition)
     try:
         with cr.savepoint():
             cr.execute(query)
+            cr.execute(
+                'COMMENT ON CONSTRAINT "{}" ON "{}" IS %s'.format(constraintname, tablename), (definition,))
```

Comment:
Isn't here the possibility of SQL injection?

Reply:
Indeed @rco-odoo you should probably use psycopg2.sql.Identifier everywhere in this file (t.i.l. AsIs is not safe either). See https://stackoverflow.com/a/42947632

Reply:
@pedrobaeza you could consider an SQL injection if the constraint definition was given by some unchecked program input. But that's not the case: the constraint definition is in your program itself.

Reply:
@StefanRijnhart may I suggest to define your query as the one above?
```python
query1 = 'ALTER TABLE...'.format(...)
query2 = 'COMMENT ON CONSTRAINT...'.format(...)
with cr.savepoint():
    cr.execute(query1)
    cr.execute(query2, [definition])
```

Reply:
Done